### ISSUE TITLE
Implement lensing related functions

### ISSUE THREAD

**Issue by pennalima** [LSSTDESC/CCL#791]
Implement convergence, shear, reduced shear and magnification.
These functions are required by CLMM (Cluster WG). 
We are including CCL as a back-end to compute the cosmological/modeling functions.

**Comment by damonge** [comment]
Hi @pennalima 

Some of these are already present in CCL (see [here](https://ccl.readthedocs.io/en/latest/api/pyccl.halos.profiles.html#pyccl.halos.profiles.HaloProfile.projected)), and it should be easy to extend the current formalism to any other profile projections.

If accuracy is a concern (judging from the CCL telecon minutes), I would note that:
- For the NFW profile we have analytic formulas for most of the profile-related quantities, so those should be exact (it may be that you have to select the analytic calculation to avoid using FFTLog by default, but the results are there).
- For all other profiles, things will be computed by default with FFTLog. If that's not accurate enough, we can probably come up with more accurate integrators (or rather, use the ones we have already implemented at the C level) or other strategies.

It'd be good to know exactly what is needed (e.g. is NFW enough? What specific additional calculations do we need to implement?). Is there a document we could look at?

**Comment by damonge** [comment]
Having a set of benchmarks would also be great.

**Comment by pennalima** [comment]
Hi @damonge, 

I just shared in desc-ccl slack channel a pdf of the notebook where we compare some functions of the CCL file you mentioned (profiles.py)  with NumCosmo and Colossus. 
The functions we need to implement are written in terms of the  projected and cumul2d functions (besides cosmological destances $\Sigma_{crit}$).
We will use NFW, Einasto and DK14 profiles (this last I need to implement both in CCL and NumCosmo). 

We checked that the analytical NFW computations are ok.
